conf/config_validator.go

```diff
@@ -107,3 +134,88 @@ func (o ValidationErrors) Error() string {
 
 	return errString
 }
+
+// jsonToConfigContext takes a json config and a context string and returns a copy
+// of the context string with the field names instead of list indexes.
+// For example, if the context is "namespaces.0", and the json config is:
+//
+//	{
+//	  "namespaces": [
+//	    {
+//	      "name": "test"
+//	    }
+//	  ]
+//	}
+//
+// The function will return "namespaces.test"
+// An error is returned if the json or context are not in the expected management lib format.
+func jsonToConfigContext(jsonConfig any, context string) (string, error) {
+	split := strings.SplitN(context, ".", 2)
+	key := split[0]
+
+	var res string
+
+	// check if key is an index
+	if index, err := strconv.Atoi(key); err == nil {
```

Comment:
The only possible edge case I can think of is the `logging` context. In the schema `logging` is an array. Will this work as expected in that case?

Reply:
`logging` works, see below
```shell
./bin/asconfig validate -a 6.4.0 testdata/cases/server70/server70.conf
context: logging.console
        - description: Additional property drv-mem is not allowed, error-type: additional_property_not_allowed
```
logging.0 goes to logging.console in this case. It works because the function only expects the json config to be a slice at the point where an index is found. When `key` == "logging" jsonConfig is the map containing "logging", then jsonConfig is set the logging slice, and key becomes "0" at which point jsonConfig is asserted as a slice.